### PR TITLE
DA-800 update some fields for participant summary

### PR DESCRIPTION
**Pull request by Seeker1911**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,8 +225,8 @@ The summary includes the following fields:
 * `enrollmentStatus`
 * `race`
 * `physicalMeasurementsStatus`: indicates whether this participant has completed physical measurements
-* `physicalMeasurementsFinalizedTime`: indicates the first time physical measurements were finalized for the participant
-* `physicalMeasurementsTime`: indicates the first time physical measurements were submitted for the participant
+* `physicalMeasurementsFinalizedTime`: indicates the latest time physical measurements were finalized for the participant
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply README.md]: Yes, the doc was wrong.

File: rest-api/dao/biobank_order_dao.py [hunk 2]
```diff
@@ -233,17 +233,18 @@ def _get_non_cancelled_biobank_orders(self, session, participantId):
       BiobankOrder.created).all()
 
   def _refresh_participant_summary(self, session, obj):
-    # called when cancelled or amendments (maybe restore)
+    # called when cancelled/restored
     participant_summary_dao = ParticipantSummaryDao()
     participant_summary = participant_summary_dao.get_for_update(session, obj.participantId)
     non_cancelled_orders = self._get_non_cancelled_biobank_orders(session, obj.participantId)
+    if obj.orderStatus == BiobankOrderStatus.CANCELLED:
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/dao/biobank_order_dao.py]: ah, missed that. Thanks. Returned to original logic.

File: rest-api/test/test_data.py [hunk 3]
```diff
@@ -37,9 +37,13 @@ def data_path(filename):
   return os.path.join(os.path.dirname(__file__), 'test-data', filename)
 
 
-def load_measurement_json(participant_id, now=None):
+def load_measurement_json(participant_id, now=None, alternate=False):
   """Loads a PhysicalMeasurement FHIR resource returns it as parsed JSON."""
-  with open(data_path('measurements-as-fhir.json')) as measurements_file:
+  if alternate:
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/test_data.py]: done.

File: rest-api/test/unit_test/api_test/participant_summary_api_test.py [hunk 4]
```diff
@@ -1380,6 +1380,33 @@ def test_physical_measurement_status(self):
     # status should still be completed because participant has another valid PM
     self.assertEquals('COMPLETED', ps_1.get('physicalMeasurementsStatus'))
 
+  def test_participant_summary_returns_latest_pm(self):
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/participant_summary_api_test.py]: the pm/bio tests have a more complete restore and cancel test suite. This one is a little out of place, what would you like to see different ?

File: rest-api/test/unit_test/api_test/physical_measurements_api_test.py [hunk 5]
```diff
@@ -263,17 +264,22 @@ def test_cancel_a_physical_measuremnet(self):
     self.send_post(path, measurement2)
     path = path + '/' + response['id']
     cancel_info = get_restore_or_cancel_info()
+    ps = self.send_get('ParticipantSummary?participantId=%s' % _id)
+    print ps['entry']
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/physical_measurements_api_test.py]: done

File: rest-api/dao/physical_measurements_dao.py [hunk 6]
```diff
@@ -370,11 +371,16 @@ def _do_update_with_patch(self, session, measurement, resource):
       measurement.cancelledSiteId = site_id
       measurement.cancelledTime = clock.CLOCK.now()
       measurement.status = PhysicalMeasurementsStatus.CANCELLED
+      measurement.finalized = None
```
Comment (calbach) [inline rest-api/dao/physical_measurements_dao.py]: Should finalizedSiteId / finalizedUsername also be cleared here?

File: rest-api/test/unit_test/api_test/participant_summary_api_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (calbach) [reply rest-api/test/unit_test/api_test/participant_summary_api_test.py]: Eh, it's probably not necessary. Maybe would be nice if there were a test here which covered a cancelled PM and checked the effects on the summary.

File: rest-api/dao/physical_measurements_dao.py (hunk 6 above)
Comment (Seeker1911) [reply rest-api/dao/physical_measurements_dao.py]: The problem with physical measurements is they have no history like other tables, so I don't believe we can get them back if the order is restored later. Unless we want to take the approach that who ever restores it, becomes the finalized site/author, which I considered. Thoughts ?

File: rest-api/test/unit_test/api_test/participant_summary_api_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/participant_summary_api_test.py]: I added participant summary checking in the physical measurement tests (6 in one, half-dozen in the other 😄 )

File: rest-api/dao/physical_measurements_dao.py (hunk 6 above)
Comment (calbach) [reply rest-api/dao/physical_measurements_dao.py]: Got it - what you are describing sounds reasonable. Checking though - is that the current behavior in this PR?

When a restore happens - won't it overwrite the finalized{userName,siteId} at L381 ? Or is that somehow getting pulled out of the database?

File: rest-api/test/unit_test/api_test/participant_summary_api_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (calbach) [reply rest-api/test/unit_test/api_test/participant_summary_api_test.py]: Thanks

File: rest-api/dao/physical_measurements_dao.py (hunk 6 above)
Comment (Seeker1911) [reply rest-api/dao/physical_measurements_dao.py]: you're right, they will be overwritten by current site_id/author of the request. Maybe we should rethink that logic later, but at least for now, the values are not None for cancelled.

**Pull request merged**